src/main/java/network/brightspots/rcv/DominionCvrReader.java

```diff
@@ -315,23 +315,25 @@ private int parseCvrFile(
             }
             Integer candidateId = (Integer) rankingMap.get("CandidateId");
             String candidateCode = candidateId.toString();
-            Set<String> candidates = contestIdToCandidateCodes.get(contestId);
-            if (!candidates.contains(candidateCode)) {
+            String candidateName = config.getNameForCandidate(candidateCode);
+            Set<String> candidates = contestIdToCandidateNames.get(contestId);
+            if (!candidates.contains(candidateName)) {
               Logger.severe(
-                  "Candidate code '%s' is not valid for contest '%d'!", candidateCode,
+                  "Candidate name '%s' is not valid for contest '%d'!", candidateName,
                   contestId);
               throw new CvrParseException();
             }
             // We also need to throw an error if this candidate doesn't appear in the tabulator's
             // config file for this contest.
             if (candidateCode.equals(undeclaredWriteInLabel)) {
-              candidateCode = Tabulator.UNDECLARED_WRITE_IN_OUTPUT_LABEL;
-            } else if (!config.getCandidateCodeList().contains(candidateCode)) {
+              candidateName = Tabulator.UNDECLARED_WRITE_IN_OUTPUT_LABEL;
+            } else if (candidateName == null) {
               unrecognizedCandidateCounts.merge(candidateCode, 1, Integer::sum);
```

Comment:
Really not sure at this point, but we still want to merge `candidateCode` here instead of `candidateName`? Only asking because I see `candidateName` being merged in `HartCvrReader`.

Reply:
`candidateName` is null here, but code is probably not